src/borg/testsuite/helpers.py

```diff
@@ -607,6 +607,19 @@ def test_get_config_dir(monkeypatch):
         assert get_config_dir() == "/var/tmp"
 
 
+def test_get_config_dir_compat(monkeypatch):
+    """test that it works the same for legacy and for non-legacy implementation"""
+    monkeypatch.delenv("BORG_CONFIG_DIR", raising=False)
+    monkeypatch.delenv("XDG_CONFIG_HOME", raising=False)
+    assert get_config_dir(legacy=False) == get_config_dir(legacy=True)
+    # TODO fails on macOS: assert '/Users/tw/Library/Preferences/borg' == '/Users/tw/.config/borg'
```

Comment:
on macOS, it looks like platformdirs uses a different default: not in `~/.config/<appname>`.

Reply:
on Linux, this works ok.

Reply:
on cygwin, this works ok.

Reply:
win32 fail: assert 'C:/Users/runneradmin/AppData/Local/borg/borg' == 'D:\\a\\_temp\\msys64\\home\\runneradmin/.config/borg'

Reply:
Looks like it is using the `HOME` env var, which inside MSYS2 is set to the home directory of the MSYS2 installation.
```py
>>> os.path.expanduser("~")
'C:/Users/Rayyan'
>>> os.environ.get("HOME")
'C:\\msys64\\home\\Rayyan'
```

Reply:
ok, i excluded that "legacy compat test" on win32, no need to have it there.

BTW, that `.../Local/borg/borg` path is correct?

Reply:
@RayyanAnsari seen this?